### DISCUSSION TITLE
Use S3 for metadata instead of etcd?

### DISCUSSION THREAD

**Discussion by jamesliu4c** [openobserve/openobserve#4268] [Ideas]
We attempted to self-host etcd, and one frustration we had was with setting up etcd in our Kubernetes cluster. Of course, we did not use the etcd that Kubernetes uses, since we are using EKS instead of self-managing.

But why not use S3 to store metadata? Of course, it will be slower than etcd, but could it be fast enough?

**Comment by prabhatsharma** [comment]
OpenObserve does not recommend using etcd anymore.

We recommend postgres for metadata.
NATS for cluster coordination.

Helm chart defaults to it.

**Reply by jamesliu4c** [reply]
Is there an option to us Postgres for cluster coordination as well? I understand that NATS has a reputation for being straightforward to manage, but it is one more thing.

**Reply by prabhatsharma** [reply]
>Is there an option to us Postgres for cluster coordination as well

No
